packages/flame/lib/src/cache/assets_cache.dart

```diff
@@ -1,16 +1,26 @@
 import 'dart:convert';
 import 'dart:typed_data';
 
-import 'package:flutter/services.dart' show rootBundle;
+import 'package:flame/flame.dart';
+import 'package:flutter/services.dart' show AssetBundle;
 
 /// A class that loads, and caches files.
 ///
 /// It automatically looks for files in the `assets` directory.
 class AssetsCache {
-  final String prefix;
-  final Map<String, _Asset<dynamic>> _files = {};
+  AssetsCache({
+    this.prefix = 'assets/',
+    AssetBundle? bundle,
+  }) {
+    this.bundle = bundle ?? Flame.bundle;
+  }
```

Comment:
```suggestion
  }) : bundle = bundle ?? Flame.bundle;
```
Like this the `bundle` doesn't have to be late :)